images/tests/eks/e2e_test.go

```diff
@@ -0,0 +1,94 @@
+//go:build e2e
+// +build e2e
+
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	ts "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestAws(t *testing.T) {
+	t.Parallel()
+
+	setupPath := "./setup/"
+	defer ts.RunTestStage(t, "teardown", func() {
+		terraform.Destroy(t, ts.LoadTerraformOptions(t, setupPath))
+	})
+	ts.RunTestStage(t, "setup", func() {
+		opts := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: setupPath,
+		})
+		ts.SaveTerraformOptions(t, setupPath, opts)
+		terraform.InitAndApply(t, opts)
+	})
+	setupOpts := ts.LoadTerraformOptions(t, setupPath)
+
+	// Group tests that require an EKS cluster to exists
+	t.Run("eks", func(t *testing.T) {
```

Comment:
seems like a nice grouping of all the `eks` components we want to test that can share the same setup.

this can be adopted in a similar way for things like common `gcp` or `az` setup